semtypes/src/main/java/io/ballerina/types/XmlPrimitive.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.types;
+
+/**
+ * Each bit represents a XML singleton subtype or a empty sequence.
+ * `primitives` fields are composed of allowed values from this singleton set.
+ * Atom of the `Bdd` represented by `sequence` field is also composed of these to indicate
+ * non-empty sequence of allowed singletons.
+ *
+ * @since 3.0.0
+ */
+public class XmlPrimitive {
+    public static final int XML_PRIMITIVE_NEVER      = 1;
```

Comment:
Since these doesn't have a meaning of their own, shall we move these to XmlSubtypeData class